src/main/ruby/core/truffle/io_operations.rb

```diff
@@ -16,6 +16,18 @@ def self.set_last_line(value, a_binding)
       Truffle::KernelOperations.frame_local_variable_set(:$_, a_binding, value)
     end
 
+    def self.print(io, args, last_line_binding)
+      if args.empty?
+        raise 'last_line_binding is required' if last_line_binding.nil?
```

Comment:
Minor: I find `unless last_line_binding` more readable than explicit `nil?` checks (unless we need to differentiate `false`/`nil` which is rare).